packages/template-retail-react-app/app/components/product-view/index.jsx

```diff
@@ -641,12 +665,134 @@ const ProductView = forwardRef(
+                                                          },
+                                                          {
+                                                              storeName: (
+                                                                  <Link
+                                                                      as="button"
+                                                                      color="blue.600"
+                                                                      textDecoration="underline"
+                                                                      onClick={onOpenStoreLocator}
+                                                                  >
+                                                                      {storeName}
+                                                                  </Link>
+                                                              )
+                                                          }
+                                                      )}
+                                            </Text>
+                                        )}
+                                        {/* Show label if pickup is disabled due to no store/inventoryId */}
+                                        {!pickupEnabled && !storeName && !inventoryId && (
+                                            <Text
+                                                color="black"
+                                                fontWeight={600}
+                                                mb={3}
+                                                data-testid="pickup-select-store-msg"
+                                            >
+                                                <FormattedMessage
+                                                    defaultMessage="Pickup in "
+                                                    id="product_view.label.pickup_in_select_store_prefix"
+                                                />{' '}
+                                                <Link
+                                                    as="button"
+                                                    color="blue.600"
+                                                    textDecoration="underline"
+                                                    onClick={onOpenStoreLocator}
+                                                >
+                                                    <FormattedMessage
+                                                        defaultMessage="Select Store"
+                                                        id="product_view.label.select_store_link"
+                                                    />
+                                                </Link>
+                                            </Text>
+                                        )}
+                                    </>
+                                )}
```

Comment:
Can we break this into its own component (delivery-option)? This will make it easier to strip the BOPIS feature during app generation, also make customization easier.